engine/src/cmd/damageable.cpp

```diff
@@ -430,3 +438,31 @@ float Damageable::currentTotalShieldVal( const Shield &shield )
     return maxshield;
 }
 
+bool Damageable::flickerDamage()
+{
+    float damagelevel = GetHullPercent();
+    static double counter = getNewTime();
```

Comment:
the types should be normalized so it's all working with floats or doubles so that things don't get lost in the mix-ups. Would prefer doubles.

Reply:
It should, but it's important to understand I'm refactoring the overall structure of the code, not the inside of specific functions (unless I have to). This is all "original" code. If I tried to refactor every function in unit class and its sub-classes, the next time you'd hear from me would be a decade from now.
I'd also be very careful about switching to double everywhere. I researched it before and there's some code in save game I think that reads a double and then converts it to two floats or some similar kludge. Exactly the sort of thing to generate the subtle bugs we all love.

Reply:
yes, we'll get those things fixed in time. Thanks!